libraries/apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/api/NormalizedCache.kt

```diff
@@ -146,5 +147,12 @@ abstract class NormalizedCache : ReadOnlyNormalizedCache {
 
     private val specialChars = "()^$.*?+{}"
   }
+
+  /**
+   * Closes resources associated with the cache if any.
+   * This function must not call `nextCache.close()`, this is done by the caller.
+   */
+  override fun close() {
+  }
```

Comment:
I went with `SqlNormalizedCache.close()` and not `SqlNormalizedCacheFactory.close()` because I think it makes more sense? Ultimately, `SqlNormalizedCacheFactory` may be turned into a plain Kotlin function like we have `() -> OkHttpClient` for lazy initialization, we could have `() -> NormalizedCache`, this is the same pattern.

Reply:
Yes, that's more intuitive than a closeable factory 👍 

Now if a driver is passed, or a factory is re-used, `apolloStore.close()` must be called with caution. Should we call that out, maybe in the KDoc of the `SqlNormalizedCacheFactory(driver: SqlDriver)` constructor?